spark-commons/src/main/scala/za/co/absa/spark/commons/errorhandling/types/ColumnOrValue.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.errorhandling.types
+
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.functions.{array, col, lit, map_from_arrays, typedLit}
+import org.apache.spark.sql.types.{DataType, StringType}
+import za.co.absa.spark.commons.sql.functions.null_col
+
+import scala.language.higherKinds
+
+trait ColumnOrValue[T] {
+  def column: Column
+  def isColumn: Boolean
+  def isValue: Boolean
+  def columnNames: Set[String]
+  def getValue: Option[T]
+}
+
+object ColumnOrValue {
+  type CoV[T] = ColumnOrValue[T] //just a shorthand
+  val CoV: ColumnOrValue.type = ColumnOrValue
+
+  def apply[T](columnName: String): ColumnOrValue[T] = CoVNamedColumn(columnName)
+  def apply[T](column: Column): ColumnOrValue[T] = CoVDefinedColumn(column)
+  def apply[T](mapColumnNames: Set[String], columnTransformer: ColumnTransformer): ColumnOrValue[Map[String, T]] = CoVMapColumn(mapColumnNames, columnTransformer) //should it be explicit function?
```

Comment:
line too long but otherwise I think that it doesn't necessarily have to be an explicit function